src/app/components/fees/fee-item.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState } from 'react';
+
+import { AnimatePresence, motion } from 'framer-motion';
+import { Stack } from 'leather-styles/jsx';
+
+import { Button, ItemLayout } from '@leather.io/ui';
+
+import type { FeeType } from '@app/common/fees/use-fees';
+
+import { FormError } from '../error/form-error';
+import { FeeItemIcon } from './fee-item-icon';
+
+export interface FeeItemProps {
+  feeType: FeeType;
+  isSelected: boolean;
+  isInsufficientBalance: boolean;
+  onSelect(feeType: FeeType): void;
+  titleLeft: string;
+  captionLeft: string;
+  titleRight?: string;
+  captionRight?: string;
+}
+
+export function FeeItem({
+  feeType,
+  onSelect,
+  isInsufficientBalance,
+  isSelected,
+  titleLeft,
+  captionLeft,
+  titleRight,
+  captionRight,
+}: FeeItemProps) {
+  const [isTouched, setIsTouched] = useState(false);
+  const showInsufficientBalanceError = isTouched && isInsufficientBalance;
+
+  return (
+    <Button
```

Comment:
Maybe we can add a `FeeButton` wrapper to avoid repeating some stuff e.g. `Add margin compensation when not selected to maintain consistent size`

Reply:
OK 👍 it just seemed like a lot of repeated stuff at first glance, fair enough 👍